services/packagist/packagist-php-version.service.js

```diff
@@ -66,11 +66,58 @@ export default class PackagistPhpVersion extends BasePackagistService {
     }
   }
 
-  transform({ json, user, repo, version = '' }) {
-    const packageVersion =
-      version === ''
-        ? this.getDefaultBranch(json, user, repo)
-        : json.packages[this.getPackageName(user, repo)][version]
+  findVersionIndex(json, version) {
+    return json.findIndex(v => v.version === version)
+  }
+
+  async findSpecifiedVersion(json, user, repo, version, server) {
+    let release
+
+    if ((release = json[this.findVersionIndex(json, version)])) {
+      return release
+    } else {
+      try {
+        const allData = await this.fetchDev({
+          user,
+          repo,
+          schema: allVersionsSchema,
+          server,
+        })
+
+        const versions = BasePackagistService.expandPackageVersions(
+          allData,
+          this.getPackageName(user, repo)
+        )
+
+        return versions[this.findVersionIndex(versions, version)]
+      } catch (e) {
+        return release
+      }
+    }
+  }
```

Comment:
I believe my mental context is fresh enough at the moment to follow this, but if anyone is receptive and willing to do so in a follow up PR, I think a brief inline comment summarizing the what/why would be helpful for future readers (likely myself included)